### PR TITLE
Bug 2027929: test/e2e/upgrade/adminack: Poll gates for duration of update

### PR DESCRIPTION
**Pull request by wking**
Picking #26649 back to 4.9, but manually because I don't have #26650 in 4.9.

### CODE REVIEW

**Comment by wking** [comment]
/hold

`master`/4.10 will also need to drop the current skip.

**Comment by wking** [comment]
I've pushed 83d48587ba -> ac8b7905c6, rebasing on 4.9's current tip and picking up #26661 and #26668.  The current mapping from 4.10 -> 4.9 commits is:

* 05374fd501 -> ac8b7905c6 Bug 2026806: clusterversionoperator/adminack.go: Check for nil cm map
* 25d7ccd5c8 -> 415403d92f test/extended/util/openshift/clusterversionoperator: Drop waitForUpgradeable
* e5d0eaeda7 -> b186659968 test/extended/util/openshift/clusterversionoperator: Return errors
* 32c8f30e9a -> 8246a6abd2 test/extended/util/openshift/clusterversionoperator: Simplify Fail(f) calls
* 4d1a25aee9 -> 5095fb3ec8 test/e2e/upgrade/adminack: Poll gates for duration of update

No significant manual work, although there were some minor conflicts to resolve, because 4.9 lacks #26650.

File: test/extended/util/adminack.go [hunk 1]
```diff
@@ -36,6 +37,24 @@ var adminAckGateRegexp = regexp.MustCompile(adminAckGateFmt)
 // admin-acks configmap to ack the given admin-ack gate. Once all gates have been ack'ed, the test waits for the
 // Upgradeable condition to change to true.
```
Comment (jottofar) [inline test/extended/util/adminack.go]: Replace with:
`// If the gate has been ack'ed the test verifies that the Upgradeable condition does not complain about the ack. Test`
`// then clears the ack and verifies that the Upgradeable condition complains about the ack. Test then sets the ack`
`// and verifies that the Upgradeable condition no longer complains about the ack.`

**Comment by wking** [comment]
there's still [one nit][1] with the 4.10 test-case, but I think we can punt that one to follow-up work and take this PR back as it stands.

/hold cancel

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=2026806#c8

**Pull request merged**